test/demo/findIntent.inttest.ts

```diff
@@ -0,0 +1,38 @@
+import 'jest';
```

Comment:
Cool. Worth doing a pass to bring the new tests more in line with the existing tests

Reply:
The existing tests seem to be testing app specific (such as listener-related behaviour etc) whereas these ones are testing the API behaviour directly on app-directory. Couldn't think of a nice way to make them look very similar, I am open for some suggestions with this one :)

Reply:
I think the first and last test in the "without context" block are essentially doing the same thing, and even if you want to keep both, they both belong in the 'Without ad-hoc apps running' block

Reply:
Though, looking at those tests, it does raise questions about how we want this new behaviour to interact with `addIntentListener`.

I feels like it should be the case that `findIntent(intent)` will always return more results than `findIntent(intent, context)` (so context is refining the results, never expanding them), but that isn't necessarily the case with the current implementation.

Grab me when you're at a good point to discuss?

Reply:
Try working through `getApplicationsForIntent`, imagining we have a single app, "Dialer". Imagine it's in the app directory with the "fdc3.DialCall" intent and no contexts, and it's running, but hasn't added an intent listener.

Let's say we call `getApplicationsForIntent(''fdc3.DialCall", "any-context")`.
- `directoryAppsWithIntent` will contain "Dialer"
- Since we've specified a context `directoryAppsWithIntent` will be returned as is, so the result contains "Dialer"

Let's say we call `getApplicationsForIntent(''fdc3.DialCall", "any-context")`.
- `allAppWindows` will contain "Dialer" (because "Dialer" is running)
- `directoryAppsWithIntent` will contain "Dialer"
- Since we've not specified the context, the execution will take the `else` branch
- `appsInModelWithIntent` **won't** contain "Dialer" (because "Dialer" hasn't added an intent listener)
- `directoryAppsNotInModel` **won't** contain "Dailer" (because "Dialer" *is* in the model)
- So our return result `[...appsInModelWithIntent, ...directoryAppsNotInModel]` won't contain "Dialer"

So this is a case where specifying the context grows rather than shrinks what gets returned. Have I got anything wrong here?

Reply:
Fully agree that `getApplicationsForIntent(intent, context)` should be a strict subset of `getApplicationsForIntent(intent)` - in the former, can we "validate" the results we get from the directory against the model, and remove any where the app hasn't (yet?) registered listeners for intent types that it claimed support for within the directory entry?

Reply:
I think "`directoryAppsNotInModel` won't contain "Dialer" (because "Dialer" is in the model)" is not correct, because `directoryAppsNotInModel` is a difference set of directory apps and `appsInModelWithIntent`. 

Given "Dialer" is not included in `appsInModelWithIntent` it will be included in `directoryAppsNotInModel`

Reply:
I am suggesting to change to this:
```            const directoryAppsNotInModel = directoryAppsWithIntent
                .filter(directoryApp => !appsInModelWithIntent.some(app => app.appId === directoryApp.appId));```

Reply:
So that's a change in behaviour of the "without-context" case, which previously would not have returned "Dialer", but now will do. And so I think the test "When calling findIntent with the intent, the app is NOT returned" will fail.

I think it's more we want to reduce what gets returned in the "with-context" case, not expand what gets returned in the "without-context" case.

Unless you're intentionally proposing that we change the existing behaviour here.

Reply:
@bryangaleOF Is there a specific reason why we don't want to return "Dialer" in "without-context" case if it declares "fdc3.DialCall" as an intent? I thought the idea of `getApplicationsForIntent` was to return all the application that matches the specified intent (and context if eligible)

Reply:
Since Dialer is running, and we know it hasn't registered any intent listeners, we know in practice that it's unable to handle the the intent. That's given precedence over what's in the app directory

Reply:
oh, gotcha!

Reply:
👍